src/main/java/org/jabref/gui/externalfiles/LinkedFileHandler.java

```diff
@@ -0,0 +1,144 @@
+package org.jabref.gui.externalfiles;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.jabref.logic.util.io.FileUtil;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.LinkedFile;
+import org.jabref.model.metadata.FilePreferences;
+import org.jabref.model.util.FileHelper;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LinkedFileHandler {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LinkedFileHandler.class);
+
+    private final BibDatabaseContext databaseContext;
+    private final FilePreferences filePreferences;
+    private final BibEntry entry;
+
+    private final LinkedFile fileEntry;
+
+    public LinkedFileHandler(LinkedFile fileEntry, BibEntry entry, BibDatabaseContext databaseContext, FilePreferences filePreferences) {
+        this.fileEntry = fileEntry;
+        this.entry = entry;
+        this.databaseContext = Objects.requireNonNull(databaseContext);
+        this.filePreferences = Objects.requireNonNull(filePreferences);
+    }
+
+    public boolean moveToDefaultDirectory() throws IOException {
```

Comment:
I don't understand why you duplicate the code of the moveFilesCleanup, is not necessary, as the moveFiles cleanup can also work with a single linked file

Reply:
Ah I see you refactored this out of the cleanup, But you now have a architecture violation from logic to gui